src/uct/ib/base/ib_iface.c

```diff
@@ -1307,10 +1342,9 @@ uct_ib_iface_init_gid_info(uct_ib_iface_t *iface,
             goto out;
         }
     } else {
-        gid_info->gid_index             = (cfg_gid_index ==
-                                           UCS_ULUNITS_AUTO) ?
-                                          UCT_IB_MD_DEFAULT_GID_INDEX :
-                                          cfg_gid_index;
+        gid_info->gid_index             = (cfg_gid_index != UCS_ULUNITS_AUTO) ?
+                                          cfg_gid_index :
```

Comment:
maybe unite this logic between IB/ROCE - that if cfg did index is not auto, use it

Reply:
tried that looks heavier than it was. 
we will have to check iface type (roce or not) at least twice, also will need to change `uct_ib_iface_init_roce_gid_info` and `uct_ib_device_select_gid` to unify with ib part

Reply:
maybe move the AUTO check to uct_ib_iface_gid_index?